cli/src/commands/run.rs

```diff
@@ -59,7 +59,7 @@ pub fn cmd_run(ui: &mut Ui, command: &CommandHelper, args: &RunArgs) -> Result<(
     // 2. the amount of cores available.
     // 3. a single job, if all of the above fails.
     let _jobs = match args.jobs {
-        Some(0) => return Err(user_error("must pass at least one job")),
+        Some(0) => std::thread::available_parallelism().map(|t| t.into()).ok(),
         Some(jobs) => Some(jobs),
         None => std::thread::available_parallelism().map(|t| t.into()).ok(),
```

Comment:
```suggestion
        Some(0) | None => std::thread::available_parallelism().map(|t| t.into()).ok(),
        Some(jobs) => Some(jobs),
```

suggestion: Can fold these cases together

Reply:
I initially did that but clippy treated it as unreachable. See https://github.com/martinvonz/jj/actions/runs/10081922624/job/27875038811

Reply:
Ok, it was a ordering problem.